### PR TITLE
Use Native Coroutines instead of gen decorator

### PR DESCRIPTION
**Pull request by WillAyd**
Unpins older tornado version with updates as suggested by Tornado:

https://www.tornadoweb.org/en/stable/guide/coroutines.html#native-vs-decorated-coroutines

### CODE REVIEW

File: tabpy-server/tabpy_server/psws/callbacks.py [hunk 1]
```diff
@@ -36,7 +35,6 @@ def wait_for_endpoint_loaded(python_service, object_uri):
         sleep(0.1)
 
 
-@gen.coroutine
```
Comment (WillAyd) [inline tabpy-server/tabpy_server/psws/callbacks.py]: Items in this module didn't appear to be doing anything asynchronous so simply removed the decorator and did not add async in contrast to other methods

**Comment by WillAyd** [comment]
Hmm so error message seems to be a result of mixing but multithreaded and asynchronous operations. Is there any particular reason to have both in the application or would a port to entirely asynchronous be considered?

File: tabpy-server/setup.py [hunk 2]
```diff
@@ -46,8 +46,7 @@
         'requests',
         'singledispatch',
         'six',
-        'tornado==5.1.1',
-        'Tornado-JSON',
+        'tornado',
```
Comment (rxing-tableau) [inline tabpy-server/setup.py]: We need to guarantee all the tests pass before upgrade tornado to the latest version.
Can't just install the latest one.

File: tabpy-server/tabpy_server/handlers/evaluation_plane_handler.py [hunk 3]
```diff
@@ -112,5 +109,4 @@ def call_subprocess(self, function_to_evaluate, arguments):
         else:
             future = self.executor.submit(_user_script, restricted_tabpy,
                                           **arguments)
-        ret = yield future
-        raise gen.Return(ret)
+        return asyncio.wrap_future(future)
```
Comment (rxing-tableau) [inline tabpy-server/tabpy_server/handlers/evaluation_plane_handler.py]: Why not ?
async def call_subprocess(self, function_to_evaluate, arguments):
...
return await future

File: tabpy-server/setup.py (hunk 2 above)
Comment (WillAyd) [reply tabpy-server/setup.py]: What version do you want to pin it to?

**Comment by WillAyd** [comment]
Thanks for the responses! I think one of them might have gotten lost in GitHub somewhere but in regards to the `call_subprocess` method I don't think it can be awaited because that actually returns a Future from a multi-threaded operation rather than an a Task.

Is there any particular reason to use multi-threaded operations or could we stick all asynchronous? Seems like the latter would make for a more unified code base with easier maintenance so happy to invest time in making that happen just not sure if there is a historical reason both paradigms were intertwined.

**Comment by rxing-tableau** [comment]
Thank you for all the contributions Will.
As I know, there is no historical blocker from moving to unified asynchronous paradigm. It just needs the refactor.
For the Tornado version, we can only support 5.1.1 now, because there is known issue when we move to higher version. It probably related with the gen.coroutines decorator.

**Comment by WillAyd** [comment]
> For the Tornado version, we can only support 5.1.1 now, because there is known issue when we move to higher version. It probably related with the gen.coroutines decorator.

Oh yea! So I should have been more explicit but that is the point of this. If we remove these decorators and place with the native async syntax from Python (as suggested by Tornado anyway) we wouldn't need to pin to 5.1. A lot of those features were intentionally removed in Tornado 6

> As I know, there is no historical blocker from moving to unified asynchronous paradigm. It just needs the refactor.

Good to know. I'll take a look at removing that altogether over the next few days then - just doing async should be easier to maintain and would be less buggy

**Comment by 0golovatyi** [comment]
@WillAyd Do you still have plans to work on this change? If not I am going to close the PR.

**Comment by WillAyd** [comment]
Feel free to close for now - can always pick up later

**Pull request closed without merging**